### PR TITLE
Add scale and transparency support

### PR DESCRIPTION
**Pull request by ItayZiv**
Scale support with global default setting, still probably needs unit testing?

### CODE REVIEW

File: sphinxcontrib/drawio.py [hunk 1]
```diff
@@ -70,8 +70,9 @@ class DrawIO(SphinxDirective):
         "height": directives.positive_int,
         "page-index": directives.nonnegative_int,
         "width": directives.positive_int,
+        "scale": directives.positive_int,
```
Comment (Modelmat) [inline sphinxcontrib/drawio.py:73]: Alphabetical order please :)

File: sphinxcontrib/drawio.py [hunk 2]
```diff
@@ -139,6 +140,9 @@ def render_drawio(self: SphinxTranslator, node: DrawIONode, in_filename: str,
             value = node["config"][option]
             extra_args.append("--{}".format(option))
             extra_args.append(str(value))
+        elif option == "scale":
```
Comment (Modelmat) [inline sphinxcontrib/drawio.py:143]: Whilst this does what you want it to do, I do not think placing it here conveys exactly what it should do.
If you have a default case, do not use optional_uniques, do it like I have done `page-index`.

File: sphinxcontrib/drawio.py [hunk 3]
```diff
@@ -268,6 +272,7 @@ def setup(app: Sphinx) -> Dict[str, Any]:
     app.add_directive("drawio", DrawIO)
     app.add_config_value("drawio_output_format", "png", "html", ENUM(*VALID_OUTPUT_FORMATS))
     app.add_config_value("drawio_binary_path", None, "html")
+    app.add_config_value("drawio_default_scale", 1, "html", directives.positive_int)
```
Comment (Modelmat) [inline sphinxcontrib/drawio.py:275]: I do not think `directives.positive_int` works here in the config values, it only supports an ENUM of output formats.

Comment (ItayZiv) [reply sphinxcontrib/drawio.py:275]: I thought I saw it taking a function/lamda? ill check again and update as needed

File: sphinxcontrib/drawio.py (hunk 1 above)
Comment (Modelmat) [reply sphinxcontrib/drawio.py:80]: I don't think this is alphabetical order :)

File: sphinxcontrib/drawio.py [hunk 4]
```diff
@@ -104,6 +113,14 @@ def render_drawio(self: SphinxTranslator, node: DrawIONode, in_filename: str,
 
     page_index = str(node["config"].get("page-index", 0))
     output_format = node["config"].get("format") or default_output_format
+    if "scale" in node["config"]:
+        scale = str(node["config"].get("scale"))
+    else:
+        scale = str(self.config.drawio_default_scale)
+    if "transparency" in node["config"]:
```
Comment (Modelmat) [inline sphinxcontrib/drawio.py:120]: Are we not doing the caching as discussed?
There is no reason to do `.get()` if you have checked it is already inside the node.
```py
scale = node["config"].get("scale")
if scale is None:
	scale = self.config.drawio_default_scale
....
"--scale",
str(scale),
```

Ditto for the transparency.

File: README.md [hunk 5]
```diff
@@ -106,4 +126,25 @@ This option allows control over the alignment of the image on the page.
...
+
+### Transparency
+- *Formal Name*: `:transparency:`
+- *Default Value*: `drawio_default_transparency` set in conf.py
+- *Possible Values*: `"true"` or `"false"`
+
+This changes the background transparency for diagrams exported to `png` files.
+Will override `drawio_default_transparency` which was st in conf.py for this
+specific diagram. If this setting is specified while the output format is not
+`png` it will have no effect on the generated image
```
Comment (Modelmat) [inline README.md:150]: please add empty last line and remove this change 
image

File: README.md [hunk 6]
```diff
@@ -106,4 +126,25 @@ This option allows control over the alignment of the image on the page.
...
-other valid pages (draw.io binary functionality)
\ No newline at end of file
+other valid pages (draw.io binary functionality)
+
+### Scale
+- *Formal Name*: `:scale:`
+- *Default Value*: `drawio_default_scale` set in conf.py
+- *Possible Values*: any positive integer
+
+This simply scales the size of the output image. By default draw.io *usually*
```
Comment (Modelmat) [inline README.md:136]: Please mention this with numbers (e.g. a nominally 50x50 image will become approximately 100x100 with a scale of 2)

File: tests/test_options.py [hunk 7]
```diff
@@ -43,3 +43,19 @@ def test_width_height(images: List[Path]):
     assert get_image_size(images[1]) == (202, 102)
     assert get_image_size(images[2]) == (1007, 511)
 
+# noinspection PyTypeChecker
```
Comment (Modelmat) [inline tests/test_options.py:46]: extra space above please

File: sphinxcontrib/drawio.py [hunk 8]
```diff
@@ -53,6 +53,13 @@ def format_spec(argument: Any) -> str:
     return directives.choice(argument, VALID_OUTPUT_FORMATS)
 
 
+def boolean_spec(argument: Any) -> bool:
```
Comment (Modelmat) [inline sphinxcontrib/drawio.py:56]:
```py
argument = argument.strip().lower()
if argument == "true":
	return True
elif argument == "false":
	return False
else:
	raise ValueError(".....")
```

File: README.md (hunk 5 above)
Comment (ItayZiv) [reply README.md:154]: No idea whats causing that change?

Comment (Modelmat) [reply README.md:154]: GitHub shows trailing space - else, could be line endings.

File: setup.py [hunk 9]
```diff
@@ -4,9 +4,9 @@
 
 # This will fail if something happens or if not in a git repository.
 # This is intentional.
-ret = subprocess.run("git describe --tags --abbrev=0", stdout=subprocess.PIPE,
-                     stderr=subprocess.PIPE, check=True, shell=True)
-version = ret.stdout.decode("utf-8").strip()
+#ret = subprocess.run("git describe --tags --abbrev=0", stdout=subprocess.PIPE,
```
Comment (Modelmat) [inline setup.py:7]: Could you undo this please?

**Pull request merged**